R/experiment.R

```diff
@@ -36,46 +41,77 @@ new_experiment <- function(parameters, obsrates, tmax, seed,
+
+  if(all(is.null(parameters), is.null(obsrates))) {
+    output <- NA
+  }
+  if (!is.na(output) &
+        (all(!is.null(parameters), nrow(parameters) >= 1) ||
+        all(!is.null(obsrates), nrow(obsrates) >= 1))) {
       stopifnot(all(is.list(output),
-                  length(output) > 1,
+                  length(output) >= 1,
                   sapply(output, is.data.frame)))
-    } else {
-      stopifnot(is.data.frame(output))
     }
-  }
 
   stopifnot(all(is.character(model), length(model) == 1) ||
               all(is.list(model), length(model) == 3))
 
   stopifnot(!is.na(as.numeric(tmax)))
   stopifnot(!is.na(as.numeric(seed)))
+
   # Generating new names:
+  oldnames <- NULL
   names_param <- names(parameters)
   names_obsrates <- names(obsrates)
-  oldnames <- c(names_param, names_obsrates)
-  param_newname <- paste0("p_", names_param)
-  obsrates_newname <- paste0("r_", names_obsrates)
-  newnames <- c(param_newname, obsrates_newname)
+  if (all(!is.null(parameters), !is.null(obsrates))) {
+    oldnames <- c(names_param, names_obsrates)
+    param_newname <- paste0("p_", names_param)
+    obsrates_newname <- paste0("r_", names_obsrates)
+    newnames <- c(param_newname, obsrates_newname)
+  } else if (!is.null(parameters)) {
+    oldnames <- names_param
+    param_newname <- paste0("p_", names_param)
+    newnames <- param_newname
+  } else if (!is.null(obsrates)) {
+    oldnames <- names_obsrates
+    obsrates_newname <- paste0("r_", names_obsrates)
+    newnames <- obsrates_newname
+  }
 
   # Dealing with dictionaries:
   if (is.null(dic_g2r)) {
-    dic_g2r <- setNames(newnames, oldnames)
-    dic_r2g <- setNames(names(dic_g2r), dic_g2r)
-
-  } else {
+    if (!is.null(oldnames)) {
+      dic_g2r <- setNames(newnames, oldnames)
+      dic_r2g <- setNames(names(dic_g2r), dic_g2r)
+    } else {
+      dic_g2r <- NULL
+      dic_r2g <- NULL
+   }
+ } else {
+  if (!is.null(oldnames)) {
     stopifnot(all(dic_g2r %in% oldnames))
     sel1 <- which(dic_g2r %in% names_param)
     sel2 <- which(dic_g2r %in% names_obsrates)
-    dic_g2r <- c(setNames(paste0("p_", dic_g2r[sel1]), names(dic_g2r[sel1])),
-                 setNames(paste0("r_", dic_g2r[sel2]), names(dic_g2r[sel2])))
-    dic_r2g <- setNames(names(dic_g2r), dic_g2r)
+#    if (all(!is.null(parameters), !is.null(obsrates))) {
+        dic_g2r <- c(setNames(paste0("p_", dic_g2r[sel1]), names(dic_g2r[sel1])),
+               setNames(paste0("r_", dic_g2r[sel2]), names(dic_g2r[sel2])))
+        dic_r2g <- setNames(names(dic_g2r), dic_g2r)
+#    } else if (!is.null(parameters)) {
+#        dic_g2r <- setNames(paste0("p_", dic_g2r[sel1]), names(dic_g2r[sel1]))
+#        dic_r2g <- setNames(names(dic_g2r), dic_g2r)
+#    } else if (!is.null(obsrates)) {
+#        dic_g2r <-  setNames(paste0("r_", dic_g2r[sel2]), names(dic_g2r[sel2]))
+#        dic_r2g <- setNames(names(dic_g2r), dic_g2r)
+#    }
+#  } else {
+#     dic_g2r <- NULL
+#      dic_r2g <- NULL
```

Comment:
If these lines are on comment maybe we can delete them

Reply:
ok